File: leetcode/169.多数元素/169.多数元素.go
package main

func majorityElement(nums []int) int {
	halfLen := int(len(nums) / 2)
	numCount := make(map[int]int, halfLen+1)
	for _, num := range nums {
		numCount[num] += 1
		count := numCount[num]
		if count > halfLen {
			return num
		}
	}
	return -1
}

File: leetcode/169.多数元素/169.多数元素.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        num_count = {}
        n = len(nums)
        half_n = n // 2
        for num in nums:
            count = num_count.get(num, 0) + 1
            if count > half_n:
                return num
            num_count[num] = count
